cpu/sam0_common/periph/flashpage.c

```diff
@@ -223,21 +230,25 @@ static void _write_page(void* dst, const void *data, size_t len, void (*cmd_writ
     _cmd_clear_page_buffer();
 
     /* write the first, unaligned bytes */
-    if (unaligned_start) {
-        *dst32++ = unaligned_pad_start(data, unaligned_start);
-        data = (uint8_t*)data + unaligned_start;
+    if (len_unaligned_start) {
+        *dst32++ = unaligned_pad_start(data, len_unaligned_start,
+                offset_unaligned_start);
+        data = (uint8_t*)data + len_unaligned_start;
     }
 
     /* copy whole words */
     const uint32_t *data32 = data;
```

Comment:
An `uint32_t *` should always be aligned to 4 byte to avoid confusion. I'm aware that `unaligned_get_u32()` is used to access data at this address, which is correct. But the pointer type should `uint8_t *` or `void *`, if you're not sure about the alignment. I'm aware that `dst32++` is more readable than `dst32 += sizeof(uint32_t)` (when `uint8_t *` is used) or even `dst32 = (uintptr_t)dst32 + sizeof(uint32_t)` (when `void *` is used). But otherwise this is undefined behavior.

Reply:
I assume you are referring to `data32` and not `dst32` as `dst32` is aligned at all times, if I didn't miss anything. Due to the alignment of the flash writes by removing the unaligned (referring to flash) beginning bytes from `data` (void pointer), data might become misaligned even if it was previously aligned.
I can remove `data32` and instead use `data` or a cast to `uint8_t`. I cannot remove the unaligned access for retrieving the `uint32_t` because there is no guarantee of alignment on the input data.
Would you be ok with these changes?

Reply:
I think you misunderstood what I meant. Please continue to use `unaligned_get_u32()`, but do not use `uint32_t *` as type. An `uint32_t *` must point to a RAM address that fulfills the alignment requirement of `uint32_t` at any point in time, as

1. This would be super confusing to reviewers, if pointers cannot be accessed via `*data32`
2. This would be undefined behavior (and compilers like to kick your ass, if they detect undefined behavior in your code)

So just change the type to either `void *` (technically the most correct) or to `uint8_t *` for all pointers to `uint32_t` that might be unaligned.

Reply:
I homogenized pointer arithmetic within this function according to your suggestions. If ok with you, I'll rebase to the current master.

Reply:
And thanks for the review! I also found the original data32 pointer confusing but did not dare to change it.

Reply:
Looks good to me :-)